sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -650,11 +650,17 @@ case object UnresolvedSeed extends LeafExpression with Unevaluable {
  * An intermediate expression to hold a resolved (nested) column. Some rules may need to undo the
  * column resolution and use this expression to keep the original column name.
  */
-case class TempResolvedColumn(child: Expression, nameParts: Seq[String]) extends UnaryExpression
+case class TempResolvedColumn(
+    child: Expression,
+    nameParts: Seq[String],
+    hasTried: Boolean = false) extends LeafExpression
   with Unevaluable {
+  // If it has been tried to be resolved but failed, mark it as unresolved so that other rules can
+  // try to resolve it again.
```

Comment:
This expression for its purpose is to hold original column name for a resolved column. So the column resolution can be undo. With this new `hasTried`, it becomes something that is resolved but also failed to resolve?

Reply:
Yes, I'll update the classdoc later. Now this expression can be used to undo column resolution, or redo it with a different priority.